splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/joins/BroadcastJoinIT.java

```diff
@@ -414,4 +441,104 @@ public void testInequalityBroadCastJoin() throws Exception {
 
         rs.close();
     }
+
+    @Test
+    public void testCharBooleanColumnsBroadCastJoin() throws Exception {
+        String sqlText = format("select count(s1.a1) from " + s1 +
+                " inner join " + s2 + " --SPLICE-PROPERTIES joinStrategy=BROADCAST,useSpark=%s \n" +
+                " on s1.b1 = s2.b2" , useSpark
+        ) ;
+        ResultSet rs = classWatcher.executeQuery(sqlText);
+
+        Assert.assertTrue("rs.next() failed", rs.next());
+        int c = rs.getInt(1);
+        Assert.assertTrue("count(s1.a1) returned incorrect number of rows:", (c == 1));
+
+        sqlText = format("select count(s1.a1) from " + s1 +
+                " inner join " + s2 + " --SPLICE-PROPERTIES joinStrategy=BROADCAST,useSpark=%s \n" +
+                " on s2.b2 = s1.b1" , useSpark
+        ) ;
+        rs = classWatcher.executeQuery(sqlText);
+
+        Assert.assertTrue("rs.next() failed", rs.next());
+        c = rs.getInt(1);
+        Assert.assertTrue("count(s1.a1) returned incorrect number of rows:", (c == 1));
+
+        rs.close();
+    }
+
+    @Test
+    public void testCharDateColumnsBroadCastJoin() throws Exception {
+        String sqlText = format("select count(s1.a1) from " + s1 +
```

Comment:
Can you add a negative test case where the char value  is not a legitimate date type? Would it be able to error out gracefully with the cast operation?

Reply:
Sure. It's an error show out.